File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/style/layers/CustomLayer.java
package org.maplibre.android.style.layers;

import androidx.annotation.Keep;

/**
 * Custom layer.
 * <p>
 * Experimental feature. Do not use.
 * </p>
 */
public class CustomLayer extends Layer {

  public CustomLayer(String id,
                     long host) {
    initialize(id, host);
  }

  /**
   * Triggers map re-paint.
   *
   * @deprecated Use {@link MapLibreMap#triggerRepaint()} instead.
   */
  @Deprecated
  @Keep
  public void update() {
  }

  @Keep
  CustomLayer(long nativePtr) {
    super(nativePtr);
  }

  @Keep
  protected native void initialize(String id, long host);

  @Override
  @Keep
  protected native void finalize() throws Throwable;

}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/style/layers/CustomLayer.kt
package org.maplibre.android.style.layers

import androidx.annotation.Keep

/**
 * Custom layer.
 *
 * Experimental feature. Do not use.
 */
class CustomLayer : Layer {
    constructor(id: String?, host: Long) {
        initialize(id, host)
    }

    @Keep
    internal constructor(nativePtr: Long) : super(nativePtr)

    /**
     * Triggers map re-paint.
     */
    @Deprecated("Use MapLibreMap.triggerRepaint() instead.")
    @Keep
    fun update() {
    }

    @Keep
    protected external fun initialize(
        id: String?,
        host: Long,
    )

    @Keep
    @Throws(Throwable::class)
    protected external override fun finalize()
}
